### ISSUE TITLE
`parallel_deriv_color` gives incorrect derivatives for ParallelGroups

### ISSUE THREAD

**Issue by timryanb** [OpenMDAO/OpenMDAO#2986]
### Description

I've been running into issues on an optimization when using `parallel_deriv_color` option where some of my constraint sensitivities come back zero even though they are non-zero at the component-level. This breaks my optimization. If the `parallel_deriv_color` flag is removed from my constraints the sensitivities come back non-zero, as expected, and the optimization continues without issue.

I've put together a toy problem below to demonstrate the issue.

When this problem is run in parallel with 2 procs (`mpirun -np 2 python bug_test.py`) it runs as expected.
But when run on 4 procs (2 procs per flight condition), the optimization fails because most of the gradients openmdao returns are incorrectly zeroed.

I've narrowed down the issue to this line of code in OpenMDAO: https://github.com/OpenMDAO/OpenMDAO/blob/master/openmdao/core/total_jac.py#L1423
Changing this line to an additive assignment (`self.J[ind, :] += row`) seems to fix the issue on my end.
I don't know if this is the best way to solve this problem internally, but hopefully provides some information.

### Example

```python
import openmdao.api as om


class SimpleAero(om.ExplicitComponent):
    """Simple aerodynamic model"""

    def initialize(self):
        self.options.declare(
            "CLwing",
            default=0.5,
            desc="Wing lift factor",
        )
        self.options.declare(
            "CLtail",
            default=0.25,
            desc="tail lift factor",
        )
        self.options.declare(
            "CDwing",
            default=0.05,
            desc="Wing drag factor",
        )
        self.options.declare(
            "CDtail",
            default=0.025,
            desc="tail drag factor",
        )

    def setup(self):
        # Inputs
        self.add_input('alpha', 0.1, desc="Angle of attack of wing")
        self.add_input('tail_angle', 0.01, desc="Angle of attack of tail")

        # Outputs
        self.add_output('L', 0.0, desc="Total lift")
        self.add_output('D', 0.0, desc="Total drag")

        # Set options
        self.CLwing = self.options["CLwing"]
        self.CDwing = self.options["CDwing"]
        self.CLtail = self.options["CLtail"]
        self.CDtail = self.options["CDtail"]

        # Use cs to approx partials
        self.declare_partials(of='*', wrt='*', method='cs')

    def compute(self, inputs, outputs):
        """ A simple surrogate for a 2 dof aero problem"""
        outputs['L'] = self.CLwing * inputs['alpha'] + self.CLtail * inputs['tail_angle']
        outputs['D'] = self.CDwing * inputs['alpha'] ** 2 + self.CDtail * inputs['tail_angle'] ** 2

# Use parallel group to solve flight conditions simultaneously
flight_conds = om.ParallelGroup()
flight_conds.add_subsystem("cruise", SimpleAero(CLwing=0.5, CDwing=0.05, CLtail=0.25, CDtail=0.025))
flight_conds.add_subsystem("maneuver", SimpleAero(CLwing=0.75, CDwing=0.25, CLtail=0.45, CDtail=0.15))

# build the model
prob = om.Problem()
prob.model.add_subsystem('flight_conditions', flight_conds)

# setup the optimization
prob.driver = om.pyOptSparseDriver(debug_print=['desvars', 'objs', 'nl_cons', 'totals'])
prob.driver.options['optimizer'] = 'SNOPT'

# Set dvs for each flight condition
prob.model.add_design_var('flight_conditions.cruise.alpha', lower=-50, upper=50)
prob.model.add_design_var('flight_conditions.cruise.tail_angle', lower=-50, upper=50)
prob.model.add_design_var('flight_conditions.maneuver.alpha', lower=-50, upper=50)
prob.model.add_design_var('flight_conditions.maneuver.tail_angle', lower=-50, upper=50)
# Use the parallel derivative option to solve lift constraint simultaneously
prob.model.add_constraint('flight_conditions.cruise.L', equals=1.0, parallel_deriv_color="lift")
prob.model.add_constraint('flight_conditions.maneuver.L', equals=2.5, parallel_deriv_color="lift")
# Set cruise drag as objective
prob.model.add_objective('flight_conditions.cruise.D')

prob.setup(mode='rev')

# run the optimization
prob.run_driver()
```
```


### OpenMDAO Version

3.27.1-dev

### Relevant environment information

_No response_

**Comment by timryanb** [comment]
Closing since this seems to be a duplicate of issue #2919

**Comment by naylor-b** [comment]
The underlying cause of this bug is actually different than issue #2919 so I'm reopening it.
